gimie/io.py

```diff
@@ -4,102 +4,101 @@
 import os
 from pathlib import Path
 import requests
-from typing import Optional, Union
+from typing import Iterator, Optional, Union
 
 
 class Resource:
-    """Abstract class for buffered read-only access to local or remote resources via
-    a file-like interface."""
+    """Abstract class for read-only access to local or remote resources via
+    a file-like interface.
+
+    Parameters
+    ----------
+    name:
+        The name of the resource, typically the filename.
+    """
 
     name: str
 
-    def open(self) -> io.BufferedReader:
+    def open(self) -> io.RawIOBase:
         raise NotImplementedError
 
 
 class LocalResource(Resource):
-    """Providing buffered read-only access to local data.
-
-    Parameters
-    ----------
-    name: the name of the resource, typically the filename.
-    url: the URL where the resource. can be downladed from.
-    headers: optional headers to pass to the request.
+    """Providing read-only access to local data via a file-like interface.
 
     Examples
     --------
-    >>> from gimie.io import LocalResource
     >>> resource = LocalResource("README.md")
     """
 
     def __init__(self, path: Union[str, os.PathLike]):
         self.path: Path = Path(path)
 
-    def open(self, mode="r") -> io.BufferedReader:
-        return io.BufferedReader(io.FileIO(self.path, mode))
+    def open(self, mode="r") -> io.RawIOBase:
```

Comment:
Since a `Resource` is read-only, I would not expose the `mode`. If you need it anyway, I would suggest to move it in the constructor, in order to preserve the signature of `open`. The documentation should be updated accordingly.

Reply:
Good point! addressed in [70b95b8](https://github.com/SDSC-ORD/gimie/pull/93/commits/70b95b875f36663e9f8ade65ee7e7a793bf54d51)